docs/content/docs/deep/componets.md

```diff
@@ -0,0 +1,113 @@
+# Components
+
+Fluid Framework powered applications have several ways to modularize their code and data into separate components.
+
+## Containers
+
+We expect most Fluid applications to to operate on a single container at a time, ans store all their persisted content in that container.
+It is however possible to use multiple containers, and some use-cases may require it.
+Specifically data which require different permissions, or to be stored in different locations, has to be split into separate containers.
+
+When splitting documents/data across multiple containers, it is up to the application to decide when to delete containers which are no longer needed.
+
+Each container is independent synchronization wise and there is no way to do anything atomically or even strongly ordered across multiple containers.
+
+Each container gets its own service client and thus separate op stream, storage and permissions.
+
+If desired, an application can open a second copy of a container as a mechanism to have a separate "branch" (independent view which can have edits speculatively applied).
+This is currently not well optimized (for performance, memory use or ergonomics) and the functionality of this is rather limited (its impractical to use it to preview changes, and keep rebasing them to stay up to date before committing them for example).
+
+An application using multiple containers can (though for bundle size generally shouldn't) use different versions or copies of the framework and its runtime for different containers.
```

Comment:
This only applies to one of our programming models. In the encapsulated model we it is common to use different version across out compat boundaries, primarily loader and runtime. It is also common to load different containers via a codeLoader which can also have different versions

Reply:
the statement might need more clarity around whether it's talking about heterogenous versions of different Fluid layers within a single container vs. heterogeneous versions of FF when looking at multiple containers, i.e. difference between:

container 1: runtime@2, loader@1, driver@2
container 2: runtime@2, loader@1, driver@2

and

container 1: runtime@2, loader@2, driver@2
container 2: runtime@1, loader@1, driver@1

I read the statement here as though it was talking about the latter case, but it is a bit ambiguous